WARNING:torch.distributed.run:
*****************************************
Setting OMP_NUM_THREADS environment variable for each process to be 1 in default, to avoid your system being overloaded, please further tune the variable for optimal performance in your application as needed. 
*****************************************
[ 2023-10-12 02:27:05 ] Completed Start                                                                        0.000 ms,         0.00 s total
[ 2023-10-12 02:27:05 ] Completed importing TimestampedTimer                                                   0.021 ms,         0.00 s total
Field "model_server_url" has conflict with protected namespace "model_".

You may be able to resolve this warning by setting `model_config['protected_namespaces'] = ()`.
Field "model_server_url" has conflict with protected namespace "model_".

You may be able to resolve this warning by setting `model_config['protected_namespaces'] = ()`.
Field "model_server_url" has conflict with protected namespace "model_".

You may be able to resolve this warning by setting `model_config['protected_namespaces'] = ()`.
Field "model_server_url" has conflict with protected namespace "model_".

You may be able to resolve this warning by setting `model_config['protected_namespaces'] = ()`.
Field "model_server_url" has conflict with protected namespace "model_".

You may be able to resolve this warning by setting `model_config['protected_namespaces'] = ()`.
Field "model_server_url" has conflict with protected namespace "model_".

You may be able to resolve this warning by setting `model_config['protected_namespaces'] = ()`.
Valid config keys have changed in V2:
* 'schema_extra' has been renamed to 'json_schema_extra'
Valid config keys have changed in V2:
* 'schema_extra' has been renamed to 'json_schema_extra'
Valid config keys have changed in V2:
* 'schema_extra' has been renamed to 'json_schema_extra'
Valid config keys have changed in V2:
* 'schema_extra' has been renamed to 'json_schema_extra'
Valid config keys have changed in V2:
* 'schema_extra' has been renamed to 'json_schema_extra'
Valid config keys have changed in V2:
* 'schema_extra' has been renamed to 'json_schema_extra'
[ 2023-10-12 02:27:09 ] Completed importing everything else                                                3,554.138 ms,         3.55 s total
| distributed init (rank 4): env://
| distributed init (rank 5): env://
| distributed init (rank 2): env://
| distributed init (rank 0): env://
| distributed init (rank 1): env://
| distributed init (rank 3): env://
[ 2023-10-12 02:27:10 ] Completed preliminaries                                                            1,103.580 ms,         4.66 s total
monai.transforms.io.dictionary LoadImaged.__init__:image_only: Current default value of argument `image_only=False` has been deprecated since version 1.1. It will be changed to `image_only=True` in version 1.3.
monai.transforms.io.dictionary LoadImaged.__init__:image_only: Current default value of argument `image_only=False` has been deprecated since version 1.1. It will be changed to `image_only=True` in version 1.3.
monai.transforms.io.dictionary LoadImaged.__init__:image_only: Current default value of argument `image_only=False` has been deprecated since version 1.1. It will be changed to `image_only=True` in version 1.3.
monai.transforms.io.dictionary LoadImaged.__init__:image_only: Current default value of argument `image_only=False` has been deprecated since version 1.1. It will be changed to `image_only=True` in version 1.3.
monai.transforms.io.dictionary LoadImaged.__init__:image_only: Current default value of argument `image_only=False` has been deprecated since version 1.1. It will be changed to `image_only=True` in version 1.3.
monai.transforms.io.dictionary LoadImaged.__init__:image_only: Current default value of argument `image_only=False` has been deprecated since version 1.1. It will be changed to `image_only=True` in version 1.3.
<class 'monai.transforms.utility.dictionary.AddChanneld'>: Class `AddChanneld` has been deprecated since version 0.8. It will be removed in version 1.3. please use MetaTensor data type and monai.transforms.EnsureChannelFirstd instead with `channel_dim='no_channel'`.
<class 'monai.transforms.utility.dictionary.AddChanneld'>: Class `AddChanneld` has been deprecated since version 0.8. It will be removed in version 1.3. please use MetaTensor data type and monai.transforms.EnsureChannelFirstd instead with `channel_dim='no_channel'`.
<class 'monai.transforms.utility.dictionary.AddChanneld'>: Class `AddChanneld` has been deprecated since version 0.8. It will be removed in version 1.3. please use MetaTensor data type and monai.transforms.EnsureChannelFirstd instead with `channel_dim='no_channel'`.
monai.transforms.utility.dictionary EnsureChannelFirstd.__init__:meta_keys: Argument `meta_keys` has been deprecated since version 0.9. not needed if image is type `MetaTensor`.
monai.transforms.utility.dictionary EnsureChannelFirstd.__init__:meta_keys: Argument `meta_keys` has been deprecated since version 0.9. not needed if image is type `MetaTensor`.
monai.transforms.utility.dictionary EnsureChannelFirstd.__init__:meta_keys: Argument `meta_keys` has been deprecated since version 0.9. not needed if image is type `MetaTensor`.
<class 'monai.transforms.utility.dictionary.AddChanneld'>: Class `AddChanneld` has been deprecated since version 0.8. It will be removed in version 1.3. please use MetaTensor data type and monai.transforms.EnsureChannelFirstd instead with `channel_dim='no_channel'`.
<class 'monai.transforms.utility.dictionary.AddChanneld'>: Class `AddChanneld` has been deprecated since version 0.8. It will be removed in version 1.3. please use MetaTensor data type and monai.transforms.EnsureChannelFirstd instead with `channel_dim='no_channel'`.
monai.transforms.utility.dictionary EnsureChannelFirstd.__init__:meta_keys: Argument `meta_keys` has been deprecated since version 0.9. not needed if image is type `MetaTensor`.
monai.transforms.utility.dictionary EnsureChannelFirstd.__init__:meta_keys: Argument `meta_keys` has been deprecated since version 0.9. not needed if image is type `MetaTensor`.
<class 'monai.transforms.utility.dictionary.AddChanneld'>: Class `AddChanneld` has been deprecated since version 0.8. It will be removed in version 1.3. please use MetaTensor data type and monai.transforms.EnsureChannelFirstd instead with `channel_dim='no_channel'`.
monai.transforms.utility.dictionary EnsureChannelFirstd.__init__:meta_keys: Argument `meta_keys` has been deprecated since version 0.9. not needed if image is type `MetaTensor`.
[ 2023-10-12 02:27:10 ] Completed build datasets                                                              12.653 ms,         4.67 s total
[ 2023-10-12 02:27:10 ] Completed build samplers                                                               0.052 ms,         4.67 s total
[ 2023-10-12 02:27:10 ] Completed build dataloaders                                                            0.136 ms,         4.67 s total
[ 2023-10-12 02:27:10 ] Completed generator to device                                                         21.281 ms,         4.69 s total
The parameter 'pretrained' is deprecated since 0.13 and may be removed in the future, please use 'weights' instead.
Arguments other than a weight enum or `None` for 'weights' are deprecated since 0.13 and may be removed in the future. The current behavior is equivalent to passing `weights=SqueezeNet1_1_Weights.IMAGENET1K_V1`. You can also use `weights=SqueezeNet1_1_Weights.DEFAULT` to get the most up-to-date weights.
The parameter 'pretrained' is deprecated since 0.13 and may be removed in the future, please use 'weights' instead.
Arguments other than a weight enum or `None` for 'weights' are deprecated since 0.13 and may be removed in the future. The current behavior is equivalent to passing `weights=SqueezeNet1_1_Weights.IMAGENET1K_V1`. You can also use `weights=SqueezeNet1_1_Weights.DEFAULT` to get the most up-to-date weights.
The parameter 'pretrained' is deprecated since 0.13 and may be removed in the future, please use 'weights' instead.
Arguments other than a weight enum or `None` for 'weights' are deprecated since 0.13 and may be removed in the future. The current behavior is equivalent to passing `weights=SqueezeNet1_1_Weights.IMAGENET1K_V1`. You can also use `weights=SqueezeNet1_1_Weights.DEFAULT` to get the most up-to-date weights.
[ 2023-10-12 02:27:10 ] Completed discriminator to device                                                     28.522 ms,         4.72 s total
The parameter 'pretrained' is deprecated since 0.13 and may be removed in the future, please use 'weights' instead.
The parameter 'pretrained' is deprecated since 0.13 and may be removed in the future, please use 'weights' instead.
Arguments other than a weight enum or `None` for 'weights' are deprecated since 0.13 and may be removed in the future. The current behavior is equivalent to passing `weights=SqueezeNet1_1_Weights.IMAGENET1K_V1`. You can also use `weights=SqueezeNet1_1_Weights.DEFAULT` to get the most up-to-date weights.
Arguments other than a weight enum or `None` for 'weights' are deprecated since 0.13 and may be removed in the future. The current behavior is equivalent to passing `weights=SqueezeNet1_1_Weights.IMAGENET1K_V1`. You can also use `weights=SqueezeNet1_1_Weights.DEFAULT` to get the most up-to-date weights.
The parameter 'pretrained' is deprecated since 0.13 and may be removed in the future, please use 'weights' instead.
Arguments other than a weight enum or `None` for 'weights' are deprecated since 0.13 and may be removed in the future. The current behavior is equivalent to passing `weights=SqueezeNet1_1_Weights.IMAGENET1K_V1`. You can also use `weights=SqueezeNet1_1_Weights.DEFAULT` to get the most up-to-date weights.
[ 2023-10-12 02:27:10 ] Completed loss functions                                                              26.893 ms,         4.75 s total
[ 2023-10-12 02:27:10 ] Completed models prepped for distribution                                             26.793 ms,         4.77 s total
[ 2023-10-12 02:27:10 ] Completed optimizers                                                                   0.567 ms,         4.77 s total
[ 2023-10-12 02:27:10 ] Completed grad scalers                                                                 0.025 ms,         4.77 s total
[ 2023-10-12 02:27:10 ] Completed checkpoint retrieval                                                         1.692 ms,         4.78 s total


EPOCH :: 0


[ 2023-10-12 02:27:10 ] Completed Start training                                                               0.000 ms,         0.00 s total

Training / resuming epoch 0 from training step 0

[ 2023-10-12 02:27:13 ] Completed train batch 0 to device                                                  3,154.231 ms,         3.15 s total
[ 2023-10-12 02:27:13 ] Completed train batch 0 generator forward                                            292.416 ms,         3.45 s total
[ 2023-10-12 02:27:13 ] Completed train batch 0 generator loss: 0.508                                        214.059 ms,         3.66 s total
Grad strides do not match bucket view strides. This may indicate grad was not created according to the gradient layout contract, or that the param's strides changed since DDP was constructed.  This is not an error, but may impair performance.
grad.sizes() = [32, 64, 1, 1, 1], strides() = [64, 1, 64, 64, 64]
bucket_view.sizes() = [32, 64, 1, 1, 1], strides() = [64, 1, 1, 1, 1] (Triggered internally at ../torch/csrc/distributed/c10d/reducer.cpp:323.)
Grad strides do not match bucket view strides. This may indicate grad was not created according to the gradient layout contract, or that the param's strides changed since DDP was constructed.  This is not an error, but may impair performance.
grad.sizes() = [32, 64, 1, 1, 1], strides() = [64, 1, 64, 64, 64]
bucket_view.sizes() = [32, 64, 1, 1, 1], strides() = [64, 1, 1, 1, 1] (Triggered internally at ../torch/csrc/distributed/c10d/reducer.cpp:323.)
Grad strides do not match bucket view strides. This may indicate grad was not created according to the gradient layout contract, or that the param's strides changed since DDP was constructed.  This is not an error, but may impair performance.
grad.sizes() = [32, 64, 1, 1, 1], strides() = [64, 1, 64, 64, 64]
bucket_view.sizes() = [32, 64, 1, 1, 1], strides() = [64, 1, 1, 1, 1] (Triggered internally at ../torch/csrc/distributed/c10d/reducer.cpp:323.)
Grad strides do not match bucket view strides. This may indicate grad was not created according to the gradient layout contract, or that the param's strides changed since DDP was constructed.  This is not an error, but may impair performance.
grad.sizes() = [32, 64, 1, 1, 1], strides() = [64, 1, 64, 64, 64]
bucket_view.sizes() = [32, 64, 1, 1, 1], strides() = [64, 1, 1, 1, 1] (Triggered internally at ../torch/csrc/distributed/c10d/reducer.cpp:323.)
Grad strides do not match bucket view strides. This may indicate grad was not created according to the gradient layout contract, or that the param's strides changed since DDP was constructed.  This is not an error, but may impair performance.
grad.sizes() = [32, 64, 1, 1, 1], strides() = [64, 1, 64, 64, 64]
bucket_view.sizes() = [32, 64, 1, 1, 1], strides() = [64, 1, 1, 1, 1] (Triggered internally at ../torch/csrc/distributed/c10d/reducer.cpp:323.)
Grad strides do not match bucket view strides. This may indicate grad was not created according to the gradient layout contract, or that the param's strides changed since DDP was constructed.  This is not an error, but may impair performance.
grad.sizes() = [32, 64, 1, 1, 1], strides() = [64, 1, 64, 64, 64]
bucket_view.sizes() = [32, 64, 1, 1, 1], strides() = [64, 1, 1, 1, 1] (Triggered internally at ../torch/csrc/distributed/c10d/reducer.cpp:323.)
[ 2023-10-12 02:27:14 ] Completed train batch 0 generator backward                                           774.773 ms,         4.44 s total
[ 2023-10-12 02:27:14 ] Completed train batch 0 metrics update                                                30.328 ms,         4.47 s total
Traceback (most recent call last):
  File "/mnt/Client/Ctan682hia6krkvbghfcwq2mtmmnzsxa/ctactafmnaa6edh5dendtkiwzdi5fosy/isc-demos/brats-rewrite/train_cycling_gen_0.py", line 278, in <module>
    main(args, timer)
  File "/mnt/Client/Ctan682hia6krkvbghfcwq2mtmmnzsxa/ctactafmnaa6edh5dendtkiwzdi5fosy/isc-demos/brats-rewrite/train_cycling_gen_0.py", line 260, in main
    generator, timer, metrics = train_generator_one_epoch(
  File "/mnt/Client/Ctan682hia6krkvbghfcwq2mtmmnzsxa/ctactafmnaa6edh5dendtkiwzdi5fosy/isc-demos/brats-rewrite/loops_0.py", line 150, in train_generator_one_epoch
    recons_loss = metrics["train"].agg[metrics["train"].map["epoch_loss"]] / metrics["train"].agg[metrics["train"].map["train_images_seen"]]
AttributeError: 'MetricsTracker' object has no attribute 'map'
Traceback (most recent call last):
  File "/mnt/Client/Ctan682hia6krkvbghfcwq2mtmmnzsxa/ctactafmnaa6edh5dendtkiwzdi5fosy/isc-demos/brats-rewrite/train_cycling_gen_0.py", line 278, in <module>
    main(args, timer)
  File "/mnt/Client/Ctan682hia6krkvbghfcwq2mtmmnzsxa/ctactafmnaa6edh5dendtkiwzdi5fosy/isc-demos/brats-rewrite/train_cycling_gen_0.py", line 260, in main
    generator, timer, metrics = train_generator_one_epoch(
  File "/mnt/Client/Ctan682hia6krkvbghfcwq2mtmmnzsxa/ctactafmnaa6edh5dendtkiwzdi5fosy/isc-demos/brats-rewrite/loops_0.py", line 150, in train_generator_one_epoch
    recons_loss = metrics["train"].agg[metrics["train"].map["epoch_loss"]] / metrics["train"].agg[metrics["train"].map["train_images_seen"]]
AttributeError: 'MetricsTracker' object has no attribute 'map'
Traceback (most recent call last):
  File "/mnt/Client/Ctan682hia6krkvbghfcwq2mtmmnzsxa/ctactafmnaa6edh5dendtkiwzdi5fosy/isc-demos/brats-rewrite/train_cycling_gen_0.py", line 278, in <module>
    main(args, timer)
  File "/mnt/Client/Ctan682hia6krkvbghfcwq2mtmmnzsxa/ctactafmnaa6edh5dendtkiwzdi5fosy/isc-demos/brats-rewrite/train_cycling_gen_0.py", line 260, in main
    generator, timer, metrics = train_generator_one_epoch(
  File "/mnt/Client/Ctan682hia6krkvbghfcwq2mtmmnzsxa/ctactafmnaa6edh5dendtkiwzdi5fosy/isc-demos/brats-rewrite/loops_0.py", line 150, in train_generator_one_epoch
    recons_loss = metrics["train"].agg[metrics["train"].map["epoch_loss"]] / metrics["train"].agg[metrics["train"].map["train_images_seen"]]
AttributeError: 'MetricsTracker' object has no attribute 'map'
Traceback (most recent call last):
  File "/mnt/Client/Ctan682hia6krkvbghfcwq2mtmmnzsxa/ctactafmnaa6edh5dendtkiwzdi5fosy/isc-demos/brats-rewrite/train_cycling_gen_0.py", line 278, in <module>
    main(args, timer)
  File "/mnt/Client/Ctan682hia6krkvbghfcwq2mtmmnzsxa/ctactafmnaa6edh5dendtkiwzdi5fosy/isc-demos/brats-rewrite/train_cycling_gen_0.py", line 260, in main
    generator, timer, metrics = train_generator_one_epoch(
  File "/mnt/Client/Ctan682hia6krkvbghfcwq2mtmmnzsxa/ctactafmnaa6edh5dendtkiwzdi5fosy/isc-demos/brats-rewrite/loops_0.py", line 150, in train_generator_one_epoch
    recons_loss = metrics["train"].agg[metrics["train"].map["epoch_loss"]] / metrics["train"].agg[metrics["train"].map["train_images_seen"]]
AttributeError: 'MetricsTracker' object has no attribute 'map'
Traceback (most recent call last):
  File "/mnt/Client/Ctan682hia6krkvbghfcwq2mtmmnzsxa/ctactafmnaa6edh5dendtkiwzdi5fosy/isc-demos/brats-rewrite/train_cycling_gen_0.py", line 278, in <module>
    main(args, timer)
  File "/mnt/Client/Ctan682hia6krkvbghfcwq2mtmmnzsxa/ctactafmnaa6edh5dendtkiwzdi5fosy/isc-demos/brats-rewrite/train_cycling_gen_0.py", line 260, in main
    generator, timer, metrics = train_generator_one_epoch(
  File "/mnt/Client/Ctan682hia6krkvbghfcwq2mtmmnzsxa/ctactafmnaa6edh5dendtkiwzdi5fosy/isc-demos/brats-rewrite/loops_0.py", line 150, in train_generator_one_epoch
    recons_loss = metrics["train"].agg[metrics["train"].map["epoch_loss"]] / metrics["train"].agg[metrics["train"].map["train_images_seen"]]
AttributeError: 'MetricsTracker' object has no attribute 'map'
Traceback (most recent call last):
  File "/mnt/Client/Ctan682hia6krkvbghfcwq2mtmmnzsxa/ctactafmnaa6edh5dendtkiwzdi5fosy/isc-demos/brats-rewrite/train_cycling_gen_0.py", line 278, in <module>
    main(args, timer)
  File "/mnt/Client/Ctan682hia6krkvbghfcwq2mtmmnzsxa/ctactafmnaa6edh5dendtkiwzdi5fosy/isc-demos/brats-rewrite/train_cycling_gen_0.py", line 260, in main
    generator, timer, metrics = train_generator_one_epoch(
  File "/mnt/Client/Ctan682hia6krkvbghfcwq2mtmmnzsxa/ctactafmnaa6edh5dendtkiwzdi5fosy/isc-demos/brats-rewrite/loops_0.py", line 150, in train_generator_one_epoch
    recons_loss = metrics["train"].agg[metrics["train"].map["epoch_loss"]] / metrics["train"].agg[metrics["train"].map["train_images_seen"]]
AttributeError: 'MetricsTracker' object has no attribute 'map'
ERROR:torch.distributed.elastic.multiprocessing.api:failed (exitcode: 1) local_rank: 0 (pid: 1831372) of binary: /mnt/Client/Ctan682hia6krkvbghfcwq2mtmmnzsxa/ctactafmnaa6edh5dendtkiwzdi5fosy/.venv/bin/python
Traceback (most recent call last):
  File "/mnt/Client/Ctan682hia6krkvbghfcwq2mtmmnzsxa/ctactafmnaa6edh5dendtkiwzdi5fosy/.venv/bin/torchrun", line 8, in <module>
    sys.exit(main())
  File "/mnt/Client/Ctan682hia6krkvbghfcwq2mtmmnzsxa/ctactafmnaa6edh5dendtkiwzdi5fosy/.venv/lib/python3.10/site-packages/torch/distributed/elastic/multiprocessing/errors/__init__.py", line 346, in wrapper
    return f(*args, **kwargs)
  File "/mnt/Client/Ctan682hia6krkvbghfcwq2mtmmnzsxa/ctactafmnaa6edh5dendtkiwzdi5fosy/.venv/lib/python3.10/site-packages/torch/distributed/run.py", line 794, in main
    run(args)
  File "/mnt/Client/Ctan682hia6krkvbghfcwq2mtmmnzsxa/ctactafmnaa6edh5dendtkiwzdi5fosy/.venv/lib/python3.10/site-packages/torch/distributed/run.py", line 785, in run
    elastic_launch(
  File "/mnt/Client/Ctan682hia6krkvbghfcwq2mtmmnzsxa/ctactafmnaa6edh5dendtkiwzdi5fosy/.venv/lib/python3.10/site-packages/torch/distributed/launcher/api.py", line 134, in __call__
    return launch_agent(self._config, self._entrypoint, list(args))
  File "/mnt/Client/Ctan682hia6krkvbghfcwq2mtmmnzsxa/ctactafmnaa6edh5dendtkiwzdi5fosy/.venv/lib/python3.10/site-packages/torch/distributed/launcher/api.py", line 250, in launch_agent
    raise ChildFailedError(
torch.distributed.elastic.multiprocessing.errors.ChildFailedError: 
============================================================
train_cycling_gen_0.py FAILED
------------------------------------------------------------
Failures:
[1]:
  time      : 2023-10-12_02:27:19
  host      : sc9
  rank      : 1 (local_rank: 1)
  exitcode  : 1 (pid: 1831373)
  error_file: <N/A>
  traceback : To enable traceback see: https://pytorch.org/docs/stable/elastic/errors.html
[2]:
  time      : 2023-10-12_02:27:19
  host      : sc9
  rank      : 2 (local_rank: 2)
  exitcode  : 1 (pid: 1831374)
  error_file: <N/A>
  traceback : To enable traceback see: https://pytorch.org/docs/stable/elastic/errors.html
[3]:
  time      : 2023-10-12_02:27:19
  host      : sc9
  rank      : 3 (local_rank: 3)
  exitcode  : 1 (pid: 1831375)
  error_file: <N/A>
  traceback : To enable traceback see: https://pytorch.org/docs/stable/elastic/errors.html
[4]:
  time      : 2023-10-12_02:27:19
  host      : sc9
  rank      : 4 (local_rank: 4)
  exitcode  : 1 (pid: 1831376)
  error_file: <N/A>
  traceback : To enable traceback see: https://pytorch.org/docs/stable/elastic/errors.html
[5]:
  time      : 2023-10-12_02:27:19
  host      : sc9
  rank      : 5 (local_rank: 5)
  exitcode  : 1 (pid: 1831377)
  error_file: <N/A>
  traceback : To enable traceback see: https://pytorch.org/docs/stable/elastic/errors.html
------------------------------------------------------------
Root Cause (first observed failure):
[0]:
  time      : 2023-10-12_02:27:19
  host      : sc9
  rank      : 0 (local_rank: 0)
  exitcode  : 1 (pid: 1831372)
  error_file: <N/A>
  traceback : To enable traceback see: https://pytorch.org/docs/stable/elastic/errors.html
============================================================
